### PR TITLE
`keras.utils.set_random_seed` clear the global `SeedGenerator`.

### PR DESCRIPTION
**Pull request by hertschuh**
This is needed to get reproducible results with `keras.random` ops.

 Also introduced a constant for `"global_seed_generator"`.

### CODE REVIEW

File: keras/src/utils/rng_utils.py [hunk 1]
```diff
@@ -52,6 +52,8 @@ def set_random_seed(seed):
 
     # Store seed in global state so we can query it if set.
     global_state.set_global_attribute(GLOBAL_RANDOM_SEED, seed)
+    # Remove global SeedGenerator, it will be recreated from the seed.
+    global_state.set_global_attribute("global_seed_generator", None)
```
[the remark this answers is not in the document]
Comment (hertschuh) [reply keras/src/utils/rng_utils.py]: Fixed

File: keras/src/utils/rng_utils.py [hunk 2]
```diff
@@ -52,6 +53,10 @@ def set_random_seed(seed):
 
     # Store seed in global state so we can query it if set.
     global_state.set_global_attribute(GLOBAL_RANDOM_SEED, seed)
+    # Remove global SeedGenerator, it will be recreated from the seed.
```
Comment (divyashreepathihalli) [inline keras/src/utils/rng_utils.py:59]: maybe the docstring can be updated to mention that  it now resets the Keras global random generator?

File: keras/src/utils/rng_utils_test.py [hunk 3]
```diff
@@ -23,11 +17,17 @@ def get_model_output():
...
-            return model.predict(ds)
+            return model.predict(x, batch_size=16)
 
         rng_utils.set_random_seed(42)
         y1 = get_model_output()
         rng_utils.set_random_seed(42)
         y2 = get_model_output()
         self.assertAllClose(y1, y2)
+
+    def test_set_random_seed_with_global_seed_generator(self):
```
Comment (divyashreepathihalli) [inline keras/src/utils/rng_utils_test.py:39]: maybe also add a test to check that different seed produces different results.  To make sure the reset mechanism doesn't accidentally lock the generator to a single state.

File: keras/src/utils/rng_utils.py (hunk 2 above)
Comment (hertschuh) [reply keras/src/utils/rng_utils.py:59]: Done

File: keras/src/utils/rng_utils_test.py (hunk 3 above)
Comment (hertschuh) [reply keras/src/utils/rng_utils_test.py:39]: Good idea, added
